src/dmd/typesem.d

```diff
@@ -4561,6 +4561,7 @@ Expression defaultInit(Type mt, const ref Loc loc)
         case Treference:
         case Tdelegate:
         case Tclass:    return new NullExp(loc, mt);
+        case Tnoreturn: return new AssertExp(loc, IntegerExp.createBool(false));
```

Comment:
This isn't strictly correct according to DIP1034 as this will trigger upon initialization AFAICT.
Not sure how to define an initializer that triggers upon first use.